### ISSUE TITLE
Type 'PropsWithChildren<AppProps>' is not assignable to type 'PropsWithChildren<WrapperProps>'

### ISSUE THREAD

**Issue by ribeaud** [isaachinman/next-i18next#1082]
## Describe the bug
After updating the library from `8.0.1` to `8.1.2`, getting following error while running `npm run build`:
```ts
./pages/_app.tsx:113:35
Type error: Argument of type 'FunctionComponent<WrapperProps> & { getInitialProps?(context: NextPageContext<any, AnyAction>): WrapperProps | Promise<...>; }' is not assignable to parameter of type 'ComponentClass<AppProps, any> | FunctionComponent<AppProps>'.
  Type 'FunctionComponent<WrapperProps> & { getInitialProps?(context: NextPageContext<any, AnyAction>): WrapperProps | Promise<...>; }' is not assignable to type 'FunctionComponent<AppProps>'.
    Types of parameters 'props' and 'props' are incompatible.
      Type 'PropsWithChildren<AppProps>' is not assignable to type 'PropsWithChildren<WrapperProps>'.
        Type 'PropsWithChildren<AppProps>' is missing the following properties from type 'WrapperProps': initialProps, initialState
```

## Occurs in next-i18next version
_v8.1.2_. If I remember correctly, the problem is reproducible with the latest _v8.0.x_ as well.

## Steps to reproduce
Getting a **TS** error at following line of code with same message:
```ts
export default appWithTranslation(wrapper.withRedux(App));
```
On my side, `wrapper` is defined as following:
```ts
import { createWrapper } from 'next-redux-wrapper';
export function makeStore(
  context?: Context,
  preloadedState?: RootState
): Store {
  ...
}
export const wrapper = createWrapper(makeStore);
```

## Expected behaviour
`npm run build` should work on my side.

## Additional context
* **next**: _10.0.9_
* **next-redux-wrapper**: _6.0.2_
* **node**: _15.8.0_
* **npm**: _7.5.3_

**Comment by isaachinman** [comment]
Hi @ribeaud – this is most likely due to #1009, which was essentially a bug fix.

I'm not sure what your redux types look like, but you can see a working example in TypeScript (strict mode) [here](https://github.com/isaachinman/next-i18next-typescript/blob/main/pages/_app.tsx#L7).

HOCs are infamously difficult to work with in TS, but I don't think this is an issue on the `next-i18next` side. Let me know if you have any further questions, and good luck!

**Comment by ribeaud** [comment]
Hi @isaachinman. Thanks for your quick answer. Unfortunately, the _working example_ did not help that much. The commit indication was useful though. After a while, some tries and a deeper analysis, I got **TS** happy with:
```ts
export default wrapper.withRedux(appWithTranslation(App));
```
I am too inexperienced to fully understand why. But my application is back working like a charm.
Another question I would have, I had to _cast_ following calls as well due the update mentioned above (kindly notice `locale as string`):
```ts
export const getStaticProps: GetStaticProps = async ({ locale }) => ({
  props: {
    ...(await serverSideTranslations(locale as string, [
    ...
    ])),
  },
});
```
This was not necessary before. Kindest regards and have a beautiful evening.
